tests/PHPStan/Rules/PhpDoc/WrongVariableNameInVarTagRuleTest.php

```diff
@@ -356,10 +376,18 @@ public function testEmptyArrayInitWithWiderPhpDoc(bool $checkTypeAgainstNativeTy
 		$errors = !$checkTypeAgainstNativeType
 			? []
 			: [
+				[
+					'PHPDoc tag @var with type int|null is not subtype of native type null.',
+					6,
+				],
 				[
 					'PHPDoc tag @var with type int is not subtype of native type array{}.',
 					24,
 				],
+				[
+					'PHPDoc tag @var with type array is not subtype of native type array{}.',
+					29,
+				],
```

Comment:
All constant assignments are working fine. Those is imo invalid "init variable" approaches and should be reported. 

But, this is probably solvable in shouldVarTagTypeBeReported if you really insist on previous behaviour.